### DISCUSSION TITLE
Option to default to `no-dev`? (Willing to provide PR)

### DISCUSSION THREAD

**Discussion by yakatz** [composer/composer#9974] [General]
I want to see if this is acceptable before I spend time on a PR:

I would like a way on my servers to make composer default to not installing `require-dev` pacakges. It seems like the simplest way to do this would be to add a `COMPOSER_NO_DEV` environment variable, and allow specifying `--with-dev` to override it.
I see that the `--dev` command line option is marked as deprecated and will be removed in Composer 3, however, I didn't find any discussion of this change, so I apologize if this has been previously discussed.

As I said, I would be happy to write up a PR, but I was wondering if there is any more background information.

**Comment by Seldaek** [comment]
I think adding a COMPOSER_NO_DEV sounds reasonable, but I'm not sure if I'd add a --with-dev flag or just say if you set the env you can make sure to  invoke with COMPOSER_NO_DEV=0 to cancel it. I suspect places where the env is set would never really need to do with dev? If you have a mixed use case then passing --no-dev when needed is fine. That'd keep the change to a smaller scope. WDYT?

**Reply by yakatz** [reply]
Sounds good. I finally had a chance to put together a patch (https://github.com/composer/composer/pull/10262), but I am having some issues figuring out the tests.
